<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>CORR — corr • ermineR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="CORR — corr"><meta property="og:description" content="This method examines the gene expression profiles themselves,
  not the gene scores for each gene (which is how the other methods like ORA
  work). A score is computed for a gene set based on how correlated the
  expression profiles are.
This can be thought of as a measure of how well the genes in the set
  cluster together, but they need not all be in the same cluster. Thus a gene
  set that contains two coherent clusters that encompass most of the genes in
  the set will tend to get a good score (though not as good as a gene set
  that is just one big cluster).
If you are interested in gene clustering, as opposed to simply looking at
  differntial expression, this method is appropriate. If you feel limited by
  the choice of distance metrics in ermineJ, ORA would be an alternative, but
  you have to define distinct clusters of genes to do that.
One alternative use of correlation scoring is as a control for
  gene-score-based analysis. Correlated gene sets can cause spurious high
  scores, especially if the differential expression in your study is weak. To
  use this approach, you could first use gene-score-based analysis (e.g.,
  gene score resampling and then analyze the data using correlation analysis.
  If any of your gene sets have high scores in both analyses, you should look
  at the data to see if the correlation is not associated with the
  differential expression. This is a simple (but ad hoc) alternative to using
  resampling over the samples to do the gene-score-based analysis.
Method overview taken from:
  http://erminej.msl.ubc.ca/help/tutorials/running-an-analysis-correlation/"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ermineR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.3.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>CORR</h1>

    <div class="hidden name"><code>corr.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This method examines the gene expression profiles themselves,
  not the gene scores for each gene (which is how the other methods like ORA
  work). A score is computed for a gene set based on how correlated the
  expression profiles are.</p>
<p>This can be thought of as a measure of how well the genes in the set
  cluster together, but they need not all be in the same cluster. Thus a gene
  set that contains two coherent clusters that encompass most of the genes in
  the set will tend to get a good score (though not as good as a gene set
  that is just one big cluster).</p>
<p>If you are interested in gene clustering, as opposed to simply looking at
  differntial expression, this method is appropriate. If you feel limited by
  the choice of distance metrics in ermineJ, ORA would be an alternative, but
  you have to define distinct clusters of genes to do that.</p>
<p>One alternative use of correlation scoring is as a control for
  gene-score-based analysis. Correlated gene sets can cause spurious high
  scores, especially if the differential expression in your study is weak. To
  use this approach, you could first use gene-score-based analysis (e.g.,
  gene score resampling and then analyze the data using correlation analysis.
  If any of your gene sets have high scores in both analyses, you should look
  at the data to see if the correlation is not associated with the
  differential expression. This is a simple (but ad hoc) alternative to using
  resampling over the samples to do the gene-score-based analysis.</p>
<p>Method overview taken from:
  <a href="http://erminej.msl.ubc.ca/help/tutorials/running-an-analysis-correlation/" class="external-link">http://erminej.msl.ubc.ca/help/tutorials/running-an-analysis-correlation/</a></p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">corr</span><span class="op">(</span></span>
<span>  <span class="va">expression</span>,</span>
<span>  annotation <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  aspects <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Molecular Function"</span>, <span class="st">"Cellular Component"</span>, <span class="st">"Biological Process"</span><span class="op">)</span>,</span>
<span>  iterations <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>  geneReplicates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mean"</span>, <span class="st">"best"</span><span class="op">)</span>,</span>
<span>  pAdjust <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FDR"</span>, <span class="st">"Bonferroni"</span><span class="op">)</span>,</span>
<span>  geneSetDescription <span class="op">=</span> <span class="st">"Latest_GO"</span>,</span>
<span>  customGeneSets <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  minClassSize <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  maxClassSize <span class="op">=</span> <span class="fl">200</span>,</span>
<span>  output <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  return <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-expression">expression<a class="anchor" aria-label="anchor" href="#arg-expression"></a></dt>
<dd><p>A file path or a data frame. Expression data. (test = CORR only)
Necesary correlation anaylsis. See http://erminej.msl.ubc.ca/help/input-files/gene-expression-profiles/
for data format</p></dd>


<dt id="arg-annotation">annotation<a class="anchor" aria-label="anchor" href="#arg-annotation"></a></dt>
<dd><p>Annotation. A file path, a data.frame or a platform short
name (eg. GPL127). If given a platform short name it will be downloaded
from annotation repository of Pavlidis Lab (<a href="https://gemma.msl.ubc.ca/annots/" class="external-link">https://gemma.msl.ubc.ca/annots/</a>).
To get a list of available annotations, use <code><a href="listGemmaAnnotations.html">listGemmaAnnotations</a></code>.
Note that if there is a file or folder with the same name as the platform
name in the directory, that file will be read instead of getting a copy from
Pavlidis Lab. If this file isn't a valid annotation file, the function will fail.
If providing a custom annotation file, see <code><a href="makeAnnotation.html">makeAnnotation</a></code> to do it from
R or  <a href="erminej.msl.ubc.ca/help/input-files/gene-annotations/">erminej.msl.ubc.ca/help/input-files/gene-annotations/</a> to do it manually.</p>
<p>If you are providing a custom gene set, you can leave annotation as NULL</p></dd>


<dt id="arg-aspects">aspects<a class="anchor" aria-label="anchor" href="#arg-aspects"></a></dt>
<dd><p>Character vector. Which Go aspects to include in the analysis.
Can be in long form (eg. 'Molecular Function') or short form (eg. <code>c('M','C','B')</code>)</p></dd>


<dt id="arg-iterations">iterations<a class="anchor" aria-label="anchor" href="#arg-iterations"></a></dt>
<dd><p>Number of iterations. We suggest a starting value of 10000
iterations. When you decide on parameters you like, we recommend a larger
number of iterations (perhaps 200,000 or more). This is to get sufficient
precision in the p-values to make multiple-test correction work correctly.
(test = GSR CORR and precRecall methods only)</p></dd>


<dt id="arg-genereplicates">geneReplicates<a class="anchor" aria-label="anchor" href="#arg-genereplicates"></a></dt>
<dd><p>What to do when genes have multiple scores in input file
(due to multiple probes per gene)</p></dd>


<dt id="arg-padjust">pAdjust<a class="anchor" aria-label="anchor" href="#arg-padjust"></a></dt>
<dd><p>Which multiple test correction method to use. Can be "FDR" or
'Westfall-Young' (slower).</p></dd>


<dt id="arg-genesetdescription">geneSetDescription<a class="anchor" aria-label="anchor" href="#arg-genesetdescription"></a></dt>
<dd><p>"Latest_GO", a file path that leads to a GO XML or OBO file
or a URL that leads to a go ontology file that ends with rdf-xml.gz.</p>
<p>If you left annotation as NULL and provided customGeneSets, this argument is
not required and will default to NULL. Otherwise, by default it'll be set to
"Latest_GO" which downloads the latest available GO XML file. This option won't work
without an internet connection. To get a frozen file
that you can use later, see <code><a href="goToday.html">goToday</a></code>, <code><a href="goAtDate.html">goAtDate</a></code> and <code><a href="getGoDates.html">getGoDates</a></code>.
 See <a href="http://erminej.msl.ubc.ca/help/input-files/gene-set-descriptions/" class="external-link">http://erminej.msl.ubc.ca/help/input-files/gene-set-descriptions/</a>
for details.</p></dd>


<dt id="arg-customgenesets">customGeneSets<a class="anchor" aria-label="anchor" href="#arg-customgenesets"></a></dt>
<dd><p>Path to a directory that contains custom gene set files,
 paths to custom gene set files themselves or a named list of character strings.
Use this option to create your own gene sets. If you provide directory you can
specify probes or gene symbols to include in your gene sets.
See <a href="http://erminej.msl.ubc.ca/help/input-files/gene-sets/" class="external-link">http://erminej.msl.ubc.ca/help/input-files/gene-sets/</a>
for information about format for this file. If you are providing a list, only gene
symbols are accepted.</p></dd>


<dt id="arg-minclasssize">minClassSize<a class="anchor" aria-label="anchor" href="#arg-minclasssize"></a></dt>
<dd><p>minimum class size</p></dd>


<dt id="arg-maxclasssize">maxClassSize<a class="anchor" aria-label="anchor" href="#arg-maxclasssize"></a></dt>
<dd><p>maximum class size</p></dd>


<dt id="arg-output">output<a class="anchor" aria-label="anchor" href="#arg-output"></a></dt>
<dd><p>Output file name.</p></dd>


<dt id="arg-return">return<a class="anchor" aria-label="anchor" href="#arg-return"></a></dt>
<dd><p>If results should be returned. Set to FALSE if you only want a file</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A list containing a "results" component and a "details" component.
"results" is a data.frame containing the main output. The columns of this table are</p>
<ul><li><p><code>Name</code>: the name of the gene set</p></li>
<li><p><code>ID</code>: the id of the gene set</p></li>
<li><p><code>NumProbes</code>:  the number of elements (e.g. probes) in the gene set.</p></li>
<li><p><code>NumGenes</code>: the number of genes in the gene set.</p></li>
<li><p><code>RawScore</code>: the raw statistic for the gene set. For explanations see this page</p></li>
<li><p><code>Pval</code>: the p value for the gene set.</p></li>
<li><p><code>CorrectedPvalue</code>: the corrected p pvalue. See this page for more information.</p></li>
<li><p><code>MFPvalue</code>: pvalue after multifunctionality correction. Might be missing if correction was not performed.</p></li>
<li><p><code>CorrectedMFPvalue</code>: Like CorrectedPvalue, but for the multifunctionality “corrected” pvalue.</p></li>
<li><p><code>Multifunctionality</code>: How biased the genes in the set are towards multifunctional genes.</p></li>
<li><p><code>Same as</code>: a list of gene sets which have the exact same members as this one. Such gene sets are not listed anywhere else.</p></li>
<li><p><code>GeneMembers</code>: If you selected the “Include genes” option when saving, this will contain a list of the genes that are in the gene set, separated by “|”.</p></li>
</ul><p>"details" section contain settings that were used to run the analysis.</p>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Ogan Mancarci.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer></div>






  </body></html>

