<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>CORR — corr • ermineR</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>
  
  
<meta property="og:title" content="CORR — corr" />

<meta property="og:description" content="This method examines the gene expression profiles themselves,
  not the gene scores for each gene (which is how the other methods like ORA
  work). A score is computed for a gene set based on how correlated the
  expression profiles are.  
This can be thought of as a measure of how well the genes in the set
  cluster together, but they need not all be in the same cluster. Thus a gene
  set that contains two coherent clusters that encompass most of the genes in
  the set will tend to get a good score (though not as good as a gene set
  that is just one big cluster).  
If you are interested in gene clustering, as opposed to simply looking at
  differntial expression, this method is appropriate. If you feel limited by
  the choice of distance metrics in ermineJ, ORA would be an alternative, but
  you have to define distinct clusters of genes to do that.  
One alternative use of correlation scoring is as a control for
  gene-score-based analysis. Correlated gene sets can cause spurious high
  scores, especially if the differential expression in your study is weak. To
  use this approach, you could first use gene-score-based analysis (e.g.,
  gene score resampling and then analyze the data using correlation analysis.
  If any of your gene sets have high scores in both analyses, you should look
  at the data to see if the correlation is not associated with the
  differential expression. This is a simple (but ad hoc) alternative to using
  resampling over the samples to do the gene-score-based analysis.  
Method overview taken from: 
  http://erminej.msl.ubc.ca/help/tutorials/running-an-analysis-correlation/" />
<meta name="twitter:card" content="summary" />
<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">ermineR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

      <div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>CORR</h1>
    </div>

    
    <p>This method examines the gene expression profiles themselves,
  not the gene scores for each gene (which is how the other methods like ORA
  work). A score is computed for a gene set based on how correlated the
  expression profiles are.</p>  
<p>This can be thought of as a measure of how well the genes in the set
  cluster together, but they need not all be in the same cluster. Thus a gene
  set that contains two coherent clusters that encompass most of the genes in
  the set will tend to get a good score (though not as good as a gene set
  that is just one big cluster).</p>  
<p>If you are interested in gene clustering, as opposed to simply looking at
  differntial expression, this method is appropriate. If you feel limited by
  the choice of distance metrics in ermineJ, ORA would be an alternative, but
  you have to define distinct clusters of genes to do that.</p>  
<p>One alternative use of correlation scoring is as a control for
  gene-score-based analysis. Correlated gene sets can cause spurious high
  scores, especially if the differential expression in your study is weak. To
  use this approach, you could first use gene-score-based analysis (e.g.,
  gene score resampling and then analyze the data using correlation analysis.
  If any of your gene sets have high scores in both analyses, you should look
  at the data to see if the correlation is not associated with the
  differential expression. This is a simple (but ad hoc) alternative to using
  resampling over the samples to do the gene-score-based analysis.</p>  
<p>Method overview taken from: 
  <a href='http://erminej.msl.ubc.ca/help/tutorials/running-an-analysis-correlation/'>http://erminej.msl.ubc.ca/help/tutorials/running-an-analysis-correlation/</a></p>
    

    <pre class="usage"><span class='fu'>corr</span>(<span class='no'>expression</span>, <span class='kw'>annotation</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>aspects</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='st'>"Molecular Function"</span>,
  <span class='st'>"Cellular Component"</span>, <span class='st'>"Biological Process"</span>), <span class='no'>iterations</span>,
  <span class='kw'>geneReplicates</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='st'>"mean"</span>, <span class='st'>"best"</span>), <span class='kw'>pAdjust</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='st'>"FDR"</span>, <span class='st'>"Bonferroni"</span>),
  <span class='kw'>geneSetDescription</span> <span class='kw'>=</span> <span class='st'>"Latest_GO"</span>, <span class='kw'>customGeneSets</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>minClassSize</span> <span class='kw'>=</span> <span class='fl'>20</span>, <span class='kw'>maxClassSize</span> <span class='kw'>=</span> <span class='fl'>200</span>, <span class='kw'>output</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>return</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a> Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>expression</th>
      <td><p>A file path or a data frame. Expression data. (test = CORR only)
Necesary correlation anaylsis. See http://erminej.msl.ubc.ca/help/input-files/gene-expression-profiles/
for data format</p></td>
    </tr>
    <tr>
      <th>annotation</th>
      <td><p>Annotation. A file path, a data.frame or a platform short 
name (eg. GPL127). If given a platform short name it will be downloaded
from annotation repository of Pavlidis Lab (<a href='www.chibi.ubc.ca/microannots/'>www.chibi.ubc.ca/microannots/</a>). 
Note that if there is a file or folder with the same name as the platform 
name in the directory, that file will be read instead of getting a copy from 
Pavlidis Lab. If this file isn't a valid annotation file, the function will fail.
If providing a custom annotation file, see <code><a href='makeAnnotation.html'>makeAnnotation</a></code> to do it from
R or  <a href='erminej.msl.ubc.ca/help/input-files/gene-annotations/'>erminej.msl.ubc.ca/help/input-files/gene-annotations/</a> to do it manually.</p>
<p>If you are providing a custom gene set, you can leave annotation as NULL</p></td>
    </tr>
    <tr>
      <th>aspects</th>
      <td><p>Character vector. Which Go aspects to include in the analysis.
Can be in long form (eg. 'Molecular Function') or short form (eg. <code>c('M','C','B')</code>)</p></td>
    </tr>
    <tr>
      <th>iterations</th>
      <td><p>Number of iterations (test = GSR and CORR methods only)</p></td>
    </tr>
    <tr>
      <th>geneReplicates</th>
      <td><p>What to do when genes have multiple scores in input file
(due to multiple probes per gene)</p></td>
    </tr>
    <tr>
      <th>pAdjust</th>
      <td><p>Which multiple test correction method to use. Can be "FDR" or
'Westfall-Young' (slower).</p></td>
    </tr>
    <tr>
      <th>geneSetDescription</th>
      <td><p>"Latest_GO", a file path that leads to a GO XML or OBO file
or a URL that leads to a go ontology file that ends with rdf-xml.gz. Note that
this is a mandatory argument. It defaults to Latest_GO but that option won't
work if you don't have a working internet connection. To get a frozen file
that you can use later, see <code><a href='goToday.html'>goToday</a></code>, <code><a href='goAtDate.html'>goAtDate</a></code> and <code><a href='getGoDates.html'>getGoDates</a></code>.
 See <a href='http://erminej.msl.ubc.ca/help/input-files/gene-set-descriptions/'>http://erminej.msl.ubc.ca/help/input-files/gene-set-descriptions/</a>
for details.</p>
<p>Can be set to NULL if annotation is also NULL and a cutsom gene set is provided.</p></td>
    </tr>
    <tr>
      <th>customGeneSets</th>
      <td><p>Path to a directory that contains custom gene set files,
 paths to custom gene set files themselves or a named list of character strings.
Use this option to create your own gene sets. If you provide directory you can
specify probes or gene symbols to include in your gene sets. 
See <a href='http://erminej.msl.ubc.ca/help/input-files/gene-sets/'>http://erminej.msl.ubc.ca/help/input-files/gene-sets/</a>
for information about format for this file. If you are providing a list, only gene
symbols are accepted.</p></td>
    </tr>
    <tr>
      <th>minClassSize</th>
      <td><p>minimum class size</p></td>
    </tr>
    <tr>
      <th>maxClassSize</th>
      <td><p>maximum class size</p></td>
    </tr>
    <tr>
      <th>output</th>
      <td><p>Output file name.</p></td>
    </tr>
    <tr>
      <th>return</th>
      <td><p>If results should be returned. Set to FALSE if you only want a file</p></td>
    </tr>
    </table>
    

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
                </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Ogan Mancarci.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  </body>
</html>

